i18n doesn't need to be in gemspec

jdzak opened the pull request:
As far as I can tell, it's not used anywhere in the gem code.

jdzak commented:
This is needed for applications using active_support ~> 3.1 because a dependency on i18n existed in active_support, but there was no gemspec dependency defined. 

The i18n dependency [was removed](https://github.com/rails/rails/commit/1854209bb3d06bc81034dfec4ef75412700eae74#diff-b5294af75c927435ac70f9e6016bf3a0) in active_support 3.1 and then [re-added](https://github.com/rails/rails/commit/d5646e842fc0537cc1c16b7887039b14bfaf8292) in active_support 3.2.

Here's the stack trace I got after removing i18n from the gemspec and also changing active_support to 3.1.12.

```
bcdatabase$ b rake spec
/Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/bin/ruby -S rspec spec/bcdatabase/commands/encrypt_spec.rb spec/bcdatabase/commands/epass_spec.rb spec/bcdatabase/commands/gen_key_spec.rb spec/bcdatabase/version_spec.rb spec/bcdatabase_spec.rb
You don't have i18n installed in your application. Please add it to your Gemfile and run bundle install
/Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/i18n.rb:2:in `require': cannot load such file -- i18n (LoadError)
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/i18n.rb:2:in `<top (required)>'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/inflector/transliterate.rb:3:in `require'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/inflector/transliterate.rb:3:in `<top (required)>'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/string/inflections.rb:3:in `require'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/string/inflections.rb:3:in `<top (required)>'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/xml_mini.rb:2:in `require'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/xml_mini.rb:2:in `<top (required)>'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/array/conversions.rb:1:in `require'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/array/conversions.rb:1:in `<top (required)>'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/array.rb:4:in `require'
  from /Users/jdzak/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/gems/1.9.1/gems/activesupport-3.1.12/lib/active_support/core_ext/array.rb:4:in `<top (required)>'
  from /Users/jdzak/Code/bcdatabase/lib/bcdatabase.rb:8:in `require'
  from /Users/jdzak/Code/bcdatabase/lib/bcdatabase.rb:8:in `<top (required)>'
  from /Users/jdzak/Code/bcdatabase/spec/spec_helper.rb:3:in `require'
  from /Users/jdzak/Code/bcdatabase/spec/spec_helper.rb:3:in `<top (required)>'
  from /Users/jdzak/Code/bcdatabase/spec/bcdatabase/commands/encrypt_spec.rb:1:in `require'
  from /Users/jdzak/Code/bcdatabase/spec/bcdatabase/commands/encrypt_spec.rb:1:in `<top (required)>'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/configuration.rb:896:in `load'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/configuration.rb:896:in `block in load_spec_files'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/configuration.rb:896:in `each'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/configuration.rb:896:in `load_spec_files'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/command_line.rb:22:in `run'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/runner.rb:80:in `run'
  from /Users/jdzak/.rvm/gems/ruby-1.9.3-p194/gems/rspec-core-2.14.8/lib/rspec/core/runner.rb:17:in `block in autorun'
```

rsutphin commented:
Thanks for checking this out, @jdzak. @betesh, is the presence of the i18n dependency causing an issue for you?

betesh commented:
Yes.  I'm getting a depracation warning from i18n.  See https://github.com/svenfuchs/i18n/blob/master/lib/i18n.rb#L362

It's annoying but not catastrophic.  I could add a line of configuration for i18n, but considering I don't use i18n anywhere in my code, I'd rather not.

If for the next gem you release, your gemspec requires active_support >= 3.2, then i18n would not need to be in the Gemfile, if I understand @jdzak correctly.  Considering 4.0.4 is out already, I don't think that will cause any problems for other users.  If it does, they can just continue using the current version of bcdatabase.  That's just my suggestion.

jdzak commented:
@betesh Ya, I can sympathize with you wanting to keep your app clean and lean. Right now, it looks like even if I remove the i18n dependency the gemspec, there is still an indirect dependency of i18n via active_support. 

@rsutphin @betesh  It looks like active_support only needed for Hash#extract_options!. Do you know if there are any other well maintained gems that have a similar method with less dependencies?

rsutphin commented:
If `Hash#extract_options!` is really all we're using, I'd be fine with inlining it in the library.

Commit:
Removing activesupport dependency. #21

I18n (dependency of activesupport) prints a deprecation message
when enforce_available_locales is not configured, which is annoying
for apps not using I18n. Since we are only using activesupports'
Array#extract_options! method we decided to inline it into bcdatabase
and remove the activesupport dependency.

jdzak commented:
@rsutphin @betesh Created a pull request with the proposed change. Comments appreciated.

jdzak commented:
@rsutphin Makes sense, thanks. I updated the code.

rsutphin commented:
Looks good to me.

jdzak commented:
@rsutphin I took the repetitive comment out.

jdzak commented:
@betesh Version 1.2.4 has been released with this change in it. Thanks again.
